src/docdb/tree/DocDBDocumentTreeItem.ts

```diff
@@ -33,7 +34,9 @@ export class DocDBDocumentTreeItem implements IAzureTreeItem {
     }
 
     public get label(): string {
-        return this.document.id;
+        const presentFields = documentDefaultFields.filter(element => this._document.hasOwnProperty(element));
+        const canonicalField = presentFields.find((element) => this._document[element] && this._document[element].toString() && !this._document[element].toString().startsWith("[object"));
```

Comment:
The three conditions in the find : 
1. Removes values being false-y (null, undefined) 
2. Removes cases where the toString is blank ""
3. Removes cases where `this._document[element]` is an Object which does not have it's own implementation. In such cases, the toString value returned is one of `[object Object]`, `[object null]`, `[object undefined]` etc. 

A simple type check does not work because `typeof <ObjectID>` is `object`  ¯\\_(ツ)_/¯